projects/owa-data/scripts/conversion/VPT/vpt_to_owamcap.py

```diff
@@ -203,15 +206,65 @@ def process_single_file(jsonl_file_path):
             dy = int(round(dy))
 
             if dx != 0 or dy != 0:
-                # NOTE: we suppose the mouse is pinned to the center. it takes VPT_MOUSE_PIN_NS for the program to pin the mouse to the center
-                topic = "mouse"
-                event = MouseEvent(event_type="move", x=center_x + dx, y=center_y + dy)
-                writer.write_message(event, topic=topic, timestamp=log_time - VPT_MOUSE_PIN_NS)
-
                 topic = "mouse"
-                event = MouseEvent(event_type="move", x=center_x, y=center_y)
+                event = RawMouseEvent(
+                    dx=dx,
+                    dy=dy,
+                    button_flags=RawMouseEvent.ButtonFlags.RI_MOUSE_NOP,
+                    timestamp=log_time,
+                )
                 writer.write_message(event, topic=topic, timestamp=log_time)
 
+            # mouse buttons : https://github.com/openai/Video-Pre-Training/blob/4ea1e8e0eddcdd5ae3cc88621a80c434f22b7f3d/run_inverse_dynamics_model.py#L114-L123
+            # 0 : left click, 1 : right click, 2 : middle click
+            current_tick_buttons = tick["mouse"]["buttons"]
+
+            # press buttons that are in the current tick, and not already pressed
+            for button in current_tick_buttons:
+                if button in button_state:
+                    continue # already pressed
+                else:
+                    button_state.add(button)
+                    topic = "mouse"
+                    if button == 0:  # left click
+                        button_flags = RawMouseEvent.ButtonFlags.RI_MOUSE_LEFT_BUTTON_DOWN
+                    elif button == 1:  # right click
+                        button_flags = RawMouseEvent.ButtonFlags.RI_MOUSE_RIGHT_BUTTON_DOWN
+                    elif button == 2:  # middle click
+                        button_flags = RawMouseEvent.ButtonFlags.RI_MOUSE_MIDDLE_BUTTON_DOWN
+                    else:
+                        raise ValueError(f"Unknown mouse button {button} in VPT data.")
+
+                    event = RawMouseEvent(
+                        dx=0,
+                        dy=0,
+                        button_flags=button_flags,
+                        timestamp=log_time,
+                    )
+                    writer.write_message(event, topic=topic, timestamp=log_time)
+            
+            # release buttons that are not in the current tick
```

Comment:
I guess it's better to switch the order of `press buttons that are in the current tick, and not already pressed` and `release buttons that are not in the current tick`